### PR TITLE
Stabilize SqlClient tests

### LINKED ISSUE #970: IntegrationTests.SqlClientTests.SubmitTraces [FAIL] Microsoft.Data.SqlClient.SqlException (0x80131904): Login failed for user 'sa'.
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/runs/7444225072?check_suite_focus=true

Flaky? Is it a problem with the generated password?

### PR DESCRIPTION
**Pull request by RassK**
## Why

Fixes #970

## What

Stabilizes tests by waiting for first successful connection.

### CODE REVIEW

File: test/IntegrationTests/SqlServerCollection.cs [hunk 1]
```diff
@@ -72,11 +74,36 @@ private async Task<TestcontainersContainer> LaunchSqlServerContainerAsync()
...
         await container.StartAsync();
 
         return container;
     }
+
+    private bool DatabaseLoginOperation()
+    {
+        try
+        {
+            using (var connection = new SqlConnection(GetConnectionString()))
```
Comment (pellared) [inline test/IntegrationTests/SqlServerCollection.cs:90]: we can remove a little noise and inline `GetConnectionString()` 

```suggestion
            string connectionString = $"Server=127.0.0.1,{Port};User=sa;Password={Password};TrustServerCertificate=True;";
            using (var connection = new SqlConnection(connectionString))
```

**Comment by RassK** [comment]
Currently passing (reruns) 10 / 10 
image

**Review by Kielek — APPROVED** [review]

**Review by pellared — APPROVED** [review]

File: test/IntegrationTests/Helpers/TestContainers/UntilAsyncOperationIsSucceeded.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// </copyright>
+
+using System;
+using System.Threading.Tasks;
+using DotNet.Testcontainers.Configurations;
+using DotNet.Testcontainers.Containers;
+using Microsoft.Extensions.Logging;
+
+namespace IntegrationTests.Helpers.TestContainers;
+internal class UntilAsyncOperationIsSucceeded : IWaitUntil
```
Comment (pellared) [inline test/IntegrationTests/Helpers/TestContainers/UntilAsyncOperationIsSucceeded.cs:24]: nit: I think it is usually separated by a blank line

**Comment by pjanotti** [comment]
Ugh a bunch of CI failures, but :) no SQL between them https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/runs/8118847701?check_suite_focus=true#step:5:2389

The failures seem the same for Alpine: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/runs/8118847701?check_suite_focus=true#step:4:3713

Kicking again to check if they are consistent.

**Review by pjanotti — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**